caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java

```diff
@@ -92,20 +105,38 @@ public ApplicationProviderDetails getApplicationProviderDetails(final Long id) {
    * @return the applications application type.
    */
   public ApplicationType getApplicationType(final Long id) {
-
     return applicationRepository.findById(id)
         .map(applicationMapper::toApplicationType)
         .orElseThrow(() -> new CaabApiException(
             String.format("Application with id %s not found", id), HttpStatus.NOT_FOUND));
   }
 
   /**
-   * Patch an application with new provider details.
+   * Adds/Updates an application with a new correspondence address.
+   *
+   * @param id the TDS id for the application.
+   * @param address the applications updated correspondence address
+   */
+  @Transactional
+  public void putCorrespondenceAddress(
+      final Long id,
```

Comment:
Could we rename this attribute `applicationId` to make it clearer?

Reply:
amended